packages/compass-query-bar/src/components/query-bar.spec.tsx

```diff
@@ -9,6 +9,23 @@ import QueryBar from './query-bar';
 import { Provider } from 'react-redux';
 import { configureStore } from '../stores/query-bar-store';
 import { toggleQueryOptions } from '../stores/query-bar-reducer';
+import type { EditorView } from '@codemirror/view';
+
+async function wait(ms = 50): Promise<void> {
+  return new Promise((resolve) => {
+    setTimeout(resolve, ms);
+  });
+}
+
+async function waitUntilEditorIsReady(
+  editorView: EditorView
+): Promise<boolean> {
+  do {
+    await wait();
+  } while ((editorView as any).updateState !== 0);
```

Comment:
I don't think that it should matter for the test if the editor became idle or not, this is an implementation detail, there is probably a better way to write a test case and avoid using this?